src/algebra/module/basic.lean

```diff
@@ -461,9 +461,17 @@ end add_comm_group
 
 section module
 
+variables {R} [ring R] [add_comm_group M] [module R M] [no_zero_smul_divisors R M]
+
+lemma smul_right_injective {x : M} (hx : x ≠ 0) :
```

Comment:
I think R should probably be explicit here. Is M explicit for `smul_left_injective`?

Reply:
Both `R` and `M` are implicit in `smul_left_injective`. I guess it depends whether we want to treat `injective` like `iff`, that arguments inferable from applying `injective` can be left implicit.

I could also wait for #7496 to be merged, then open a PR for making the parameters for `smul_left_injective` and for `smul_right_injective` explicit.